src/rust/src/lib.rs

```diff
@@ -2,7 +2,84 @@
 // 2.0, and the BSD License. See the LICENSE file in the root of this repository
 // for complete details.
 
+/// Returns the value of the input with the most-significant-bit copied to all
+/// of the bits.
+fn duplicate_msb_to_all(a: u8) -> u8 {
+    0u8.wrapping_sub(a >> 7)
+}
+
+/// This returns 0xFF if a < b else 0x00, but does so in a constant time
+/// fashion.
```

Comment:
I'd leave a note to the source - a reference to an openssl implementation would have likely convinced me not to puzzle my way through whether this does what it says on the label. ;-)

```suggestion
/// fashion. The logic comes from openssl, compare
/// https://github.com/openssl/openssl/blob/OpenSSL_1_1_1i/include/internal/constant_time.h#L123
```

Also, are the triple-`///` intentional?

Reply:
Triple `///` are doc-comments in Rust, meaning if you compiled this with rustdoc they'd show up as docs for hte function.